plugins/services/src/js/reducers/serviceForm/FormReducers/Containers.js

```diff
@@ -0,0 +1,132 @@
+import {
+  SET,
+  ADD_ITEM,
+  REMOVE_ITEM
+} from '../../../../../../../src/js/constants/TransactionTypes';
+import {
+  combineReducers,
+  simpleReducer
+} from '../../../../../../../src/js/utils/ReducerUtil';
+import {DEFAULT_POD_CONTAINER} from '../../../constants/DefaultPod';
+import {FormReducer as endpointsFormReducer} from './Endpoints';
+import {FormReducer as multiContainerArtifacts} from './MultiContainerArtifacts';
+import {
+  FormReducer as multiContainerHealthFormReducer
+} from '../MultiContainerHealthChecks';
+
+const containerReducer = combineReducers({
+  cpus: simpleReducer('resources.cpus'),
+  mem: simpleReducer('resources.mem'),
+  disk: simpleReducer('resources.disk')
+});
+
+module.exports = {
+  FormReducer(state, {type, path = [], value}) {
+    // eslint-disable-next-line no-unused-vars
+    const [_, index, field, secondIndex, name, subField] = path;
+
+    if (!path.includes('containers')) {
+      return state;
+    }
+
+    if (this.cache == null) {
+      this.cache = [];
+    }
+
+    if (this.healthCheckState == null) {
+      this.healthCheckState = [];
+    }
+
+    if (!state) {
+      state = [];
+    }
+
+    let newState = state.slice();
+    const joinedPath = path.join('.');
+
+    if (joinedPath === 'containers') {
+      switch (type) {
+        case ADD_ITEM:
+          const name = `container-${newState.length + 1}`;
+
+          newState.push(Object.assign({}, DEFAULT_POD_CONTAINER, {name}));
+          this.cache.push({});
+          break;
+        case REMOVE_ITEM:
+          newState = newState.filter((item, index) => {
+            return index !== value;
+          });
+          this.cache = this.cache.filter((item, index) => {
+            return index !== value;
+          });
+          break;
+      }
+
+      return newState;
+    }
+
+    if (field === 'endpoints') {
+      if (newState[index].endpoints == null) {
+        newState[index].endpoints = [];
+      }
+
+      newState[index].endpoints =
+        endpointsFormReducer(newState[index].endpoints, {type, path, value});
+    }
+
+    if (field === 'healthCheck') {
+      if (this.healthCheckState[index] == null) {
+        this.healthCheckState[index] = {};
+      }
+
+      newState[index].healthCheck = multiContainerHealthFormReducer.call(
+        this.healthCheckState[index],
+        newState[index].healthCheck,
+        {type, path: path.slice(3), value}
```

Comment:
Weren't you going to pass in the names in a new array?
Is it `[index, field, subField]`?